### ISSUE TITLE
v16.0.1 (windows): Invalid device selection using '--enable'

### ISSUE THREAD

**Issue by UselessGuru** [bzminer/bzminer#261]
`Bin\BzMiner-v15.4.3\bzminer.exe -a nexa -p stratum+tcp://nexapow.auto.nicehash.com:9200 -w ####################################.########### --pool_password x -r Blackbox --no_watchdog --http_enabled 1 --http_port 4001 --enable 9:0`

**I expect only device 9:0 to be selected. All other devices should be ignored.**
As you can see all devices are getting used.

Using '--disable' works as expected.

```
*************************
**                     **
**   BzMiner v15.4.3   **
**                     **
*************************


 Loading settings from config.txt
 <Warn> An error has occured openening settings file: config.txt
 <Warn> The document root must not be followed by other values. (line: 228, 1)
}]
 ^
 <Warn> Default configuration will be used. Delete settings file and re-run to restore defaults.
 Language: English
 Working Directory: c:\Users\#####################\NemosMiner\Bin\BzMiner-v15.4.3
 Logging to: Logging disabled
 Cuda: 12.1
 Nvidia Driver: 531.79
 AMD Driver: 3516.0 (PAL,LC)
 Intel Driver: Not Found

 Thank you for choosing BzMiner ;)

 ---------------------------------------------- devices ----------------------------------------------
 | #    | name                      | free     | total  | core    | mem     | fan | pwr    | temp    |
 | 1:0  | Radeon RX 580 Series      | 3.74gb   | 3.74gb | 1234mhz | 2100mhz | 77% | 94.0w  | 69C     |
 | 9:0  | Gigabyte RX 5700 XT AORUS | 3.72gb   | 3.72gb | 1161mhz | 1836mhz | 42% | 121.0w | 60C/74C |
 | 11:0 | MSI GTX 1660 Super Gaming | 5.77gb   | 6.00gb | 1530mhz | 7000mhz | 41% | 11.0w  | 41C     |
 | 12:0 | NVIDIA GeForce GT 1030    | 250.79mb | 2.00gb | 1227mhz | 3003mhz | 61% | 0.0w   | 73C     |
 | 16:0 | NVIDIA GeForce GTX 750 Ti | 1.42gb   | 2.00gb | 1241mhz | 2750mhz | 58% | 33.3w  | 70C     |
 |      |                           |          |        |         |         |     | 259.3w |         |
 07-28 20:16:54:0969 ------------------------------------------------------------------------- v15.4.3

 [nexa] Connecting to stratum+tcp://nexapow.auto.nicehash.com:9200 as Blackbox with wallet:
 [nexa] #################################.##########
 HTTP API Listening on 0.0.0.0:4001
 Setting nexa extra nonce: 1e32544b000000cd
 [nexa] Connected to stratum+tcp://nexapow.auto.nicehash.com:9200 using stratum as Blackbox
 [nexa] dev fee: 2.0%
 [nexa] New difficulty: 3.16gh
 [nexa] GPU 11:0: Generating LUT: 2.25gb
 <Error> [nexa] GPU 12:0: Not enough available memory (216.66mb/2.00gb). Requires 2.02gb.
 <Error> [nexa] GPU 16:0: Not enough available memory (1.39gb/2.00gb). Requires 2.03gb.
 [Manual input] - Refresh Display

 ---------------------------------------------- devices ----------------------------------------------
 | #    | name                      | free     | total  | core    | mem     | fan | pwr    | temp    |
 | 1:0  | Radeon RX 580 Series      | 3.68gb   | 3.74gb | 1236mhz | 2100mhz | 77% | 93.5w  | 69C     |
 | 9:0  | Gigabyte RX 5700 XT AORUS | 3.66gb   | 3.72gb | 2009mhz | 200mhz  | 43% | 111.5w | 60C/76C |
 | 11:0 | MSI GTX 1660 Super Gaming | 3.46gb   | 6.00gb | 1530mhz | 7000mhz | 41% | 45.8w  | 46C     |
 | 12:0 | NVIDIA GeForce GT 1030    | 224.78mb | 2.00gb | 1227mhz | 3003mhz | 60% | 0.0w   | 72C     |
 | 16:0 | NVIDIA GeForce GTX 750 Ti | 1.39gb   | 2.00gb | 1241mhz | 2750mhz | 59% | 34.9w  | 70C     |
 |      |                           |          |        |         |         |     | 285.8w |         |
 07-28 20:16:56:0737 ------------------------------------------------------------------------- v15.4.3

 ------------------ nexa - stratum+tcp://nexapow.auto.nicehash.com:9200 - Blackbox -------------------
 | #    | a/r/i | cfg | tbs | tsls | eff       | pool hr | miner hr | status                         |
 | 1:0  | -/-/- | i1  | --  | --   | --        | --      | --       | Initializing                   |
 | 9:0  | -/-/- | i1  | --  | --   | --        | --      | --       | Initializing                   |
 | 11:0 | -/-/- | i1  | --  | --   | 747.19khw | --      | 34.26mh  | Mining                         |
 | 12:0 | -/-/- | i1  | --  | --   | --        | --      | --       | Stopped: Not Enough Memory     |
 | 16:0 | -/-/- | i1  | --  | --   | --        | --      | --       | Stopped: Not Enough Memory     |
 |      |       |     |     |      |           |         |          |                                |
 | smry | -/-/- |     | --  | --   | 119.87khw | --      | 34.26mh  | diff:       3.16gh             |
 |      |       |     |     |      |           |         |          | latency:    --                 |
 |      |       |     |     |      |           |         |          | est. tbs:   1.54m              |
 07-28 20:16:56:0740 -------------------- uptime: 0d 0h 0m 1s -------------------------------- v15.4.3

 [nexa] GPU 9:0: Generating LUT: 2.25gb
```

**Comment by UselessGuru** [comment]
Any news?

**Comment by UselessGuru** [comment]
Any news?

**Comment by UselessGuru** [comment]
Still the same in 16.0.1

**Comment by UselessGuru** [comment]
Still the same in 16.0.3

**Comment by bzminer** [comment]
hey @UselessGuru i'm sorry, i've been kind of bad about watching the github issues, thanks for posting a link to this on my discord.

Unfortunately right now, in bz nexa is not mineable on 2gb cards due to the LUT taking up more than 2gb. I could add a 700mb option for lower end cards though, so i'll add that to my list

**Comment by bzminer** [comment]
ok sorry again, i just re-read and totally missed the point of your issue. this does in fact look like a bug, so i will test this out for a release today

**Comment by UselessGuru** [comment]
Confirm. Fixed in 16.0.5

**Comment by bzminer** [comment]
thanks @UselessGuru for getting back so fast!
